library/xml-model/src/main/java/org/cru/godtools/xml/model/Manifest.kt

```diff
@@ -205,88 +220,85 @@ class Manifest : BaseModel, Styles {
+                        val src = getAttributeValue(null, XML_PAGES_PAGE_SRC)
+                        skipTag()
+
+                        if (src != null) {
+                            val pos = result.tractPageTasks.size
+                            result.tractPageTasks += scope.async {
+                                parseFile(src).use { TractPage(this@Manifest, pos, fileName, it) }
+                            }
                         }
-                        else -> parser.skipTag()
                     }
-                    XMLNS_ARTICLE -> when (parser.name) {
-                        XML_PAGES_AEM_IMPORT -> {
-                            parser.getAttributeValueAsUriOrNull(XML_PAGES_AEM_IMPORT_SRC)
-                                ?.let { result.aemImports.add(it) }
-                            parser.skipTag()
-                        }
-                        else -> parser.skipTag()
+                    else -> skipTag()
+                }
+                XMLNS_ARTICLE -> when (name) {
+                    XML_PAGES_AEM_IMPORT -> {
+                        getAttributeValueAsUriOrNull(XML_PAGES_AEM_IMPORT_SRC)?.let { result.aemImports += it }
+                        skipTag()
                     }
-                    else -> parser.skipTag()
+                    else -> skipTag()
                 }
+                else -> skipTag()
             }
-        }.awaitAll()
-        return@runBlocking result
+        }
     }
 
     @WorkerThread
-    private fun parseResources(parser: XmlPullParser): Map<String?, Resource> {
-        parser.require(XmlPullParser.START_TAG, XMLNS_MANIFEST, XML_RESOURCES)
+    private fun XmlPullParser.parseResources() = buildList {
+        require(XmlPullParser.START_TAG, XMLNS_MANIFEST, XML_RESOURCES)
 
-        return buildList {
-            while (parser.next() != XmlPullParser.END_TAG) {
-                if (parser.eventType != XmlPullParser.START_TAG) continue
+        while (next() != XmlPullParser.END_TAG) {
+            if (eventType != XmlPullParser.START_TAG) continue
 
-                when (parser.namespace) {
-                    XMLNS_MANIFEST -> when (parser.name) {
-                        Resource.XML_RESOURCE -> add(Resource(this@Manifest, parser))
-                        else -> parser.skipTag()
-                    }
-                    else -> parser.skipTag()
+            when (namespace) {
+                XMLNS_MANIFEST -> when (name) {
+                    Resource.XML_RESOURCE -> add(Resource(this@Manifest, this@parseResources))
+                    else -> skipTag()
                 }
+                else -> skipTag()
             }
-        }.associateBy { it.name }
+        }
     }
 
     @WorkerThread
-    private fun XmlPullParser.parseTips(parseFile: suspend (String) -> CloseableXmlPullParser) = runBlocking {
+    private fun XmlPullParser.parseTips(scope: CoroutineScope, parseFile: suspend (String) -> CloseableXmlPullParser) =
```

Comment:
kind of, `runBlocking` provides a `CoroutineScope` that can be used to launch the `async` parsing tasks. But `runBlocking` has to wait for all launched tasks to complete before it will return. This meant that all the tips could be parsed in parallel and all pages could be parsed in parallel, but you couldn't parse tips at the same time as pages.

I basically moved the 2 separate `runBlocking` calls out to a single `runBlocking` call around the entire parsing, and then passed the `CoroutineScope` from that common `runBlocking` into the `parseTips` and `parsePages` methods